### DISCUSSION TITLE
Table 'pnptc_groups' doesn't exist

### DISCUSSION THREAD

**Discussion by gangelopoulos** [akeeba/panopticon#325] [Announcements]
## Describe the bug

A clean installation of 1.0.4 brings up an error when you try to Add a Site

## Error Details

Table 'pnptc_groups' doesn't exist
[View template: ROOT/tmp/compiled_templates/16dd64d16b1ac3b1f3dadbab593c108ae34f6c34.php line 373, compiled from ROOT/ViewTemplates/Sites/form_properties.blade.php]

DataController->execute('add')
in [/src/Controller/Sites.php ](/index.php?view=sites&task=add)(line 53)
    public function execute($task)
    {
        $this->aclCheck($task);
        return parent::execute($task);
    }
    public function fixJoomlaCoreUpdateSite(): void
    {
        $this->csrfProtection();

View->loadAnyTemplate('Sites/form_properties')
in [/tmp/compiled_templates/5e253fc6d77cf7162c77a796d19b6b90f3a55be0.php ](/index.php?view=sites&task=add)(line 82)
           
 <?php echo $this->loadAnyTemplate('Main/default_serverid'); ?>
        </div>
        <div class="tab-pane show"
             id="siteTabContentProperties" role="tabpanel" aria-labelledby="siteTabProperties" tabindex="-1"
        >
            <?php echo $this->loadAnyTemplate('Sites/form_properties'); ?>
        </div>
        <div class="tab-pane show"
             id="siteTabContentUpdate" role="tabpanel" aria-labelledby="siteTabUpdate" tabindex="-1"
        >
            <?php echo $this->loadAnyTemplate('Sites/form_update'); ?>

## Environment (please complete the following information)
 - Panopticon version: 1.0.4
 - PHP version: 8.1
 - Database type [e.g. MySQL, MariaDB]: MySQL
 - Database version: 5.7.44
 - Browser [e.g. Chrome, Safari]: Independent
 - Browser version [e.g. 22]: Independent

**Comment by gangelopoulos** [comment]
The same thing happened after the installation of Panopticon version: 1.0.3

**Comment by gangelopoulos** [comment]
The same thing also happened in version 1.0.0

There was an unrecoverable error
The server returned the following error:

0 — Table 'pnptc_groups' doesn't exist
[View template: ROOT/tmp/compiled_templates/16dd64d16b1ac3b1f3dadbab593c108ae34f6c34.php line 373, compiled from ROOT/ViewTemplates/Sites/form_properties.blade.php]

**Comment by nikosdion** [comment]
MySQL 5.7.8 introduced the `JSON` column type **but** it was possible to disable it, as far as I know. If your server does not support the `JSON` column type then the creation of the `#__sites` table will indeed fail, and Panopticon will not work.

**Reply by gangelopoulos** [reply]
Thanks for the clarification. I'll give that a try and get back with an update.

**Comment by Knock-Out** [comment]
Reaching out in this discussion as it seems I'm facing a similar issue.

Post-installation, the following error occurs:

"**Model Groups could not fetch column list for the table #__groups**".

### **Error Details**

Awf\Mvc\DataModel\Exception\NoTableColumns" in /vendor/akeeba/awf/src/Mvc/DataModel.php at line 199.

### **System**

- Panapoticon version 1.0.5,
- PHP version 8.2.13,
- mysqli 5.7.44-cll-lve.

How can I resolve this issue?

**Comment by gangelopoulos** [comment]
Unfortunately I couldn't get it to work with MySQL 5.7.44
I can't find an option to enable JSON column type

**Reply by Knock-Out** [reply]
Thank you for the suggestion; perhaps this can help.

```
CREATE TABLE #_groups (
    id INT PRIMARY KEY,
    other_column VARCHAR(255),
    json_column JSON
);
```

**Reply by Knock-Out** [reply]
Since I am not familiar with the table structure and to address the subsequent title-related error, I am adding to the query:

```
CREATE TABLE #_groups (
    id INT PRIMARY KEY,
    other_column VARCHAR(255),
    json_column JSON,
    title VARCHAR(255)
);
```

**Reply by nikosdion** [reply]
JSON support is built into MySQL 5.5 and 5.7. If your database server says otherwise, it is possible your host compiled their own version stripping support of some core features. In any case, MySQL 5.7 is out of support. Ask your host to upgrade their DB server or move to a host which does not use End of Life server software in production. Why? The same reason your shouldn’t eat at a restaurant using canned pickles which expired in the 1980s.

**Comment by nikosdion** [comment]
Your problem is that your database user does not have enough permissions to execute SHOW COLUMNS. Ask your host to fix the permissions for you. 

Every other suggestion made here is **INVALID** and would cause a ton of problems in the future.

**Reply by nikosdion** [reply]
Now that I am not traveling I can explain the “would cause a ton of problems in the future” bit.

The previous suggestions would create additional columns in the table, and also not create the columns we expect.

If the columns we expect to be there are not, adding data to and reading data from the table will cause SQL errors.

Even if you just add junk (randomly named) columns you could cause a problem, either during operation (a junk column may cause data insertion to fail), or update (we perform database updates based on the detects column names and, in rare cases, the number of columns on a table).

All of these problems are hard to troubleshoot because there's no published code path which can result in them. If it's just one person with the problem it's fairly easy for me to figure out they messed with their database and tell them to fix it. If more people follow the same bad advice I suddenly see a pattern of behaviour I cannot explain and end up wasting a lot of time chasing windmills. Yes, it's happened before and it's not fun at all.

So, sorry for the all caps and bold type, I was just trying to prevent a misunderstanding from snowballing to a small crisis in the few minutes I had at a rest stop along a long drive.

**Comment by Knock-Out** [comment]
Hello @nikosdion, thank you for the additional explanations, even though they were not strictly necessary. I fully understand the current and future issues, and your illustration with the pickles (😁) has been quite clear. 

However, I have requested my hosting provider to install MySQL 8, and they have assured me that this will be done in January 2024. 

Thank you again for your availability and clarity in providing information, and I appreciate your ongoing support. If you have any further advice or suggestions, they will be welcomed. Best regards.
